pipeline/pipeline.go

```diff
@@ -87,27 +137,186 @@ func Configure(
+	pipeline.errors = append(pipeline.errors, err)
+}
+
+// // no-lint
+// func (pipeline *Pipeline) clearErrors() {
+// 	pipeline.errors = []string{}
+// }
+
+func (pipeline *Pipeline) getErrorAsOne() error {
+	if len(pipeline.errors) > 0 {
+		return errors.New("on stage recurisiveParseFields will have any of this errors: " + strings.Join(pipeline.errors, "\n"))
+	} else {
+		return nil
+	}
+}
+
+// //environment values sourcesing configuratino your structure field by functions pipeline
+// func (pipeline *Pipeline) startConfiguringByFunctions(context *structContext) error {
+// 	currentChain := pipeline.chains
+// 	for {
+// 		if err := currentChain.stageFunction.Configure(context); err != nil {
+// 			if currentChain.next != nil {
+// 				currentChain = currentChain.next
+// 			} else {
+// 				return errors.New("can not configure field: " + context.Prefix)
+// 			}
+// 		} else {
+// 			return nil
+// 		}
+// 	}
+// }
+
+func (pipeline *Pipeline) configuringValues(context *structContext) error {
+	// if config.configureFileParsed == nil {
+	// 	config.configureFileParsed = gohocon.LoadConfig(config.fileName)
+	// }
+	valueIndirect := reflect.Indirect(context.Value)
+	switch valueIndirect.Kind() {
+	case reflect.Slice, reflect.Map, reflect.Array:
+		valueGet := pipeline.chains.stageFunction.GetComplexType(context)
+		fmt.Println("[Pipeline]: Level: debug. value get from parsing slice: ", valueGet)
+		if valueGet.CheckIsValue() {
+
+			// add check for setuping valueGet in valueIndirect
+			if valueIndirect.CanSet() {
+				fmt.Println("[Pipeline]: setupe value in struct")
+				valueIndirect.Set(valueGet.Value)
+			} else {
+				return errors.New("can not set " + valueIndirect.Kind().String() + " into struct field.")
+			}
 		} else {
-			break
+			return errors.New("Level: debug. value get not setupable value: ")
 		}
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return errors.New("not supported types of unsigned integer")
+	case reflect.String, reflect.Float32, reflect.Float64, reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		valueGet := pipeline.chains.stageFunction.GetBaseType(context)
+		if valueGet.CheckIsValue() {
+			fmt.Println("[Pipeline]: Level: debug. value get from parsing "+valueGet.Value.Kind().String()+": ", valueGet)
+			if valueIndirect.CanSet() {
+				fmt.Println("[Pipeline]: setupe value")
+				valueIndirect.Set(valueGet.Value)
+			} else {
+				return errors.New("can not set " + valueIndirect.Kind().String() + " into struct field.")
+			}
+		} else {
+			return errors.New("Level: debug. value get not setupable value: ")
+		}
+	default:
+		return errors.New("not supported type for hocon parsing")
+	}
+	return nil
+}
+
+func (pipeline *Pipeline) checkValueTypeIsPointer(value reflect.Value) error {
+	if value.Kind() != reflect.Ptr {
+		return errors.New("not working without pointer types")
```

Comment:
use `does not work` instead of `not working`

Reply:
++